### PR TITLE
add guide for setting up B2C tenant

### PR DESCRIPTION
**Pull request by martinhenningjensen**

### CODE REVIEW

File: doc/guide_B2C_tenant.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+![Policy keys](images/Custom_policy_upload.png)
+
+![Policy keys](images/Custom_policy_upload_mitid.png)
+
+![Policy keys](images/Custom_policy_upload_client_credentials.png)
+
+Notice that it's not possible to test the custom policy from the Azure portal without added a web redirect in the frontnend app registration.
+
+Add a web application to the ett frontend app with redirect URL `https://jwt.ms`. Not the login can be tested directly from the Azure portal.
```
Comment (TopSwagCode) [inline doc/guide_B2C_tenant.md:178]: Should we include this? jwt.ms can only show "code" returned, that can be used to get a bearer token. So we can only see the login was successfull, but not mapping of claims was right.

File: doc/guide_B2C_tenant.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+This document contains a step-by-step guide on how to configure the ETT B2C production tenant in Azure.
+
+### Prerequisites
+
+The following prerequiresites must be addressed before starting the configuration.
+
+- __Existing tenant:__ A B2C instance must be available, the current instance was created by T/I by request.
+- __Invitation:__ Users must be invited to the new B2C tenant. After accepting the invitation, the user may use the switch workspace option in Azure to access the tenant.
+- __Access rights:__ Users must have sufficient access rights to configure the tenant. (Currently we use the `global administrator` role)
```
Comment (ckr123) [inline doc/guide_B2C_tenant.md:13]: How to get these access rights?

File: doc/guide_B2C_tenant.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+Navigate to newly created app registration. Values for tenant-id and client-id can be found on this page. These values are needed later. Click `Add a certificate or secret`.
+
+![New b2c certificate](images/App_registration_b2c_certificate.png)
+
+Click `New client secret`.
+
+![New b2c certificate](images/App_registration_b2c_certificate_new.png)
+
+Leave `Description` and `Expires` fields empty. Click `Add`.  __?!? Decide on expires value__
```
Comment (ckr123) [inline doc/guide_B2C_tenant.md:51]: Who will make this decission? Any good practises?

File: doc/guide_B2C_tenant.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+First step is to replace GUIDs and URLs in the custom profiles. The relevant GUIDs and URLs to be replaced are marked with XML comments in the files.
+
+![Policy keys](images/Custom_policy_upload.png)
+
+![Policy keys](images/Custom_policy_upload_mitid.png)
+
+![Policy keys](images/Custom_policy_upload_client_credentials.png)
+
+Notice that it's not possible to test the custom policy from the Azure portal without added a web redirect in the frontnend app registration.
```
Comment (ckr123) [inline doc/guide_B2C_tenant.md:176]: frontend*

File: doc/guide_B2C_tenant.md (hunk 1 above)
Comment (ckr123) [reply doc/guide_B2C_tenant.md:178]: I am not sure why to use jwt.ms? Is it to test the policy?

Comment (TopSwagCode) [reply doc/guide_B2C_tenant.md:178]: yes

File: doc/guide_B2C_tenant.md (hunk 2 above)
Comment (TopSwagCode) [reply doc/guide_B2C_tenant.md:13]: ask Martin :D

File: doc/guide_B2C_tenant.md (hunk 3 above)
Comment (TopSwagCode) [reply doc/guide_B2C_tenant.md:51]: Right now we are using defaults. I would like to lower this when we have automated onboarding, so externals can renew them selfs.

File: doc/guide_B2C_tenant.md (hunk 4 above)
Comment (TopSwagCode) [reply doc/guide_B2C_tenant.md:176]: Naah. He was going for frontnerd

File: doc/guide_B2C_tenant.md (hunk 3 above)
Comment (ckr123) [reply doc/guide_B2C_tenant.md:51]: Should there be a task on it?

File: doc/guide_B2C_tenant.md (hunk 1 above)
Comment (ckr123) [reply doc/guide_B2C_tenant.md:178]: Can add it to a test section?

**Review by sahma19 — COMMENTED** [review]
Hvordan vil du bevise at denne guide faktisk er korrekt?

**Comment by TopSwagCode** [comment]
> Hvordan vil du bevise at denne guide faktisk er korrekt?

We have been 2 people working on it all the way through to be sure that not 1 person knew what was happening and also to ensure that we noted down all the tasks. But you as reviewer could go through all the steps and test it :)

**Comment by TopSwagCode** [comment]
I would like to approve this :D But since I have worked on it, I don't feel like it is the right thing to do :) Have asked entire team several times to read this through.

**Comment by sahma19** [comment]
> > Hvordan vil du bevise at denne guide faktisk er korrekt?
> 
> We have been 2 people working on it all the way through to be sure that not 1 person knew what was happening and also to ensure that we noted down all the tasks. But you as reviewer could go through all the steps and test it :)

Det er ikke mit job at bevise at det virker. Det er dit. Hvordan vil du bevise at det virker? Hvis Martin siger god for det, så er det fint.

Jeg prøvede at hentyde til at man kunne lave et script der automatiserede disse steps, eller et check 😉

**Review by sahma19 — APPROVED** [review]

File: doc/authorization-onboarding/readme.md [hunk 5]
```diff
@@ -50,7 +48,6 @@ to get into overview of client secrets.
 
 Here we can click on "New Client Secret", to create a a new client :P secret description, isn't that important and we will stick with default 6 months expire date for now.
```
Comment (tnickelsen) [inline doc/authorization-onboarding/readme.md:49]: Remove ':P'

File: doc/guide_B2C_tenant.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
+# Guide for configuring B2C tenant for Energy Origin
+
+## Introduction
+
+This document contains a step-by-step guide on how to configure the ETT B2C production tenant in Azure.
```
Comment (tnickelsen) [inline doc/guide_B2C_tenant.md:5]: Maybe describe when this is needed

**Review by tnickelsen — APPROVED** [review]

File: doc/guide_B2C_tenant.md (hunk 6 above)
Comment (TopSwagCode) [reply doc/guide_B2C_tenant.md:5]: added description: when onboarding new 3rd party client.

File: doc/authorization-onboarding/readme.md (hunk 5 above)
Comment (TopSwagCode) [reply doc/authorization-onboarding/readme.md]: removed and cleaned text up abit.

**Review by TopSwagCode — DISMISSED** [review]

File: doc/guide_B2C_tenant.md (hunk 6 above)
Comment (tnickelsen) [reply doc/guide_B2C_tenant.md:5]: I can't see this?

**Review by tnickelsen — APPROVED** [review]

**Pull request merged**